modules/core/src/main/java/org/apache/ignite/configuration/DistibutedThinClientConfiguration.java

```diff
@@ -0,0 +1,81 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.configuration;
+
+import static org.apache.ignite.internal.cluster.DistributedConfigurationUtils.makeUpdateListener;
+import static org.apache.ignite.internal.processors.configuration.distributed.DistributedBooleanProperty.detachedBooleanProperty;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.processors.configuration.distributed.DistributedChangeableProperty;
+import org.apache.ignite.internal.processors.configuration.distributed.DistributedConfigurationLifecycleListener;
+import org.apache.ignite.internal.processors.configuration.distributed.DistributedPropertyDispatcher;
+import org.apache.ignite.internal.processors.subscription.GridInternalSubscriptionProcessor;
+import org.apache.ignite.internal.util.future.GridFutureAdapter;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Thin client distributed configuration.
+ */
+public class DistibutedThinClientConfiguration {
+    /** */
+    private final IgniteLogger log;
+
+    /** . */
+    private final DistributedChangeableProperty<Boolean> showStackTrace =
+        detachedBooleanProperty("thinClientProperty.showStackTrace");
+
+    /** Message of baseline auto-adjust parameter was changed. */
+    private static final String PROPERTY_UPDATE_MESSAGE =
+        "ThinClientProperty parameter '%s' was changed from '%s' to '%s'";
+
+    /**
+     * @param ctx Kernal context.
+     */
+    public DistibutedThinClientConfiguration(
+        GridKernalContext ctx
+    ) {
+        log = ctx.log(DistibutedThinClientConfiguration.class);
+
+        GridInternalSubscriptionProcessor isp = ctx.internalSubscriptionProcessor();
+
+        isp.registerDistributedConfigurationListener(
+            new DistributedConfigurationLifecycleListener() {
+                @Override public void onReadyToRegister(DistributedPropertyDispatcher dispatcher) {
+                    showStackTrace.addListener(makeUpdateListener(PROPERTY_UPDATE_MESSAGE, log));
+
+                    dispatcher.registerProperties(showStackTrace);
+                }
+            }
+        );
+    }
+
+    /**
+     * @param showStack If {@code true} shows full stack trace on the client side.
+     */
+    public GridFutureAdapter<?> updateThinClientShowStackTraceAsync(boolean showStack) throws IgniteCheckedException {
+        return showStackTrace.propagateAsync(showStack);
+    }
+
+    /**
+     * @return If {@code true} shows full stack trace on the client side.
+     */
+    @Nullable public Boolean showFullStack() {
```

Comment:
1. This property should be in the existing `ThinClientConfiguration` class. Another configuration class is not needed.
2. Let's rename this to `sendServerExceptionStackTraceToClient`
3. Please make the javadoc more detailed, something like this:
```
If {@code true}, thin client response will include full stack trace when exception occurs. When {@code false}, only top level exception message is included.
```

Reply:
1. I disagree here, we have two type of setting for now: static and dynamic, dynamic has upper priority if dynamic unset - use static.

Reply:
1. I don't understand the need to make this dynamic.
2. It does not seem to be possible to configure this statically at all.

Reply:
I was wrong - it is possible to configure statically. No objections.

Reply:
@zstan please see above:

> Let's rename this to `sendServerExceptionStackTraceToClient`